src/components/LayoutTree.tsx

```diff
@@ -426,14 +560,39 @@ const TreeNode = ({
   cliffPaddingStyle: { paddingBottom: number }
   dragInProgress: boolean
   autofocusDepth: number
+  cursorIsGreaterThanPrevSibling: boolean
 } & Pick<CSSTransitionProps, 'in'>) => {
   const [y, setY] = useState(_y)
+  const [x, setX] = useState(_x)
   const fadeThoughtRef = useRef<HTMLDivElement>(null)
+
+  const isParentSwapped = useSelector(state => {
+    const allPatches = state.undoPatches
+    return allPatches?.flat().some(patch => patch.actions[0] === 'swapParent')
+  })
+
+  // true if the last action is newThought
   const isLastActionNewThought = useSelector(state => {
     const lastPatches = state.undoPatches[state.undoPatches.length - 1]
     return lastPatches?.some(patch => patch.actions[0] === 'newThought')
   })
 
+  // true if the last action is swapParent
+  const isLastActionSwapParent = useSelector(state => {
+    const lastPatches = state.undoPatches[state.undoPatches.length - 1]
+    return lastPatches?.some(patch => patch.actions[0] === 'swapParent')
+  })
```

Comment:
`isLastActionSwapParent` should probably be defined in `ThoughtNode`, since that is the only place it is used.

Reply:
This has been moved to `TreeNode`.